VideoLocker/src/main/java/org/edx/mobile/view/adapters/DiscussionPostsAdapter.java

```diff
@@ -30,13 +29,12 @@
     @ColorInt
     private final int edx_utility_success_dark;
 
-    @Inject
-    public DiscussionPostsAdapter(Context context, IEdxEnvironment environment) {
-        super(context, R.layout.row_discussion_thread, environment);
-        edx_brand_primary_base = context.getResources().getColor(R.color.edx_brand_primary_base);
-        edx_grayscale_neutral_light = context.getResources().getColor(R.color.edx_grayscale_neutral_light);
-        edx_brand_secondary_dark = context.getResources().getColor(R.color.edx_brand_secondary_dark);
-        edx_utility_success_dark = context.getResources().getColor(R.color.edx_utility_success_dark);
+    public DiscussionPostsAdapter(Fragment fragment, IEdxEnvironment environment) {
+        super(fragment.getContext(), R.layout.row_discussion_thread, environment);
```

Comment:
Passing the Fragment's Context to the super implementation defeats the purpose of taking the Fragment instead of the Context directly as a parameter, which was to not store the Context itself which might become stale after Activity reinstantiation. As a workaround, you could pass the application Context to super, and get the actual Context to use from that Fragment.